### PR TITLE
Augment Pathname documentation

### PR DESCRIPTION
**Pull request by cpurdy**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

The changes that I am submitting are documentation only. I attempted to maintain the style of documentation that I have seen in the Ruby core libraries, as well as in the Homebrew extensions thereof. I carefully followed the guidelines from [RubyDoc](https://en.wikibooks.org/wiki/Ruby_Programming/RubyDoc), but unfortunately I do not have a local toolchain that allows me to review the final HTML result, which I would have liked to do before submitting a change. (I am willing to set up such a toolchain if required, but I don't know where to start; the various guides are all for building casks and formula.)

I included a brief description of the changes in the commit message. Is that what this question refers to: "_Have you added an explanation of what your changes do and why you'd like us to include them?_"

My purpose in expanding this documentation was to assist future formula developers from making some of the mistakes that I made, and going down some of the rabbit holes that I went down, and to (hopefully) quickly direct them to the correct solution for what they are attempting to do.

### CODE REVIEW

File: Library/Homebrew/extend/pathname.rb [hunk 1]
```diff
@@ -354,6 +354,14 @@ def uninstall_info
   end
 
   # Writes an exec script in this folder for each target pathname.
+  #
+  # If the +targets+ require environment variables to be set by the exec script,
+  # then use #write_env_script.
+  #
+  # To +exec java+, use #write_jar_script; it will correctly configure the
+  # environment so that +JAVA_HOME+ is configured _and used to execute_ the correct
+  # version of \Java. (+JAVA_HOME+ is not configured in the Homebrew CI environment,
```
Comment (EricFromCanada) [inline Library/Homebrew/extend/pathname.rb:363]: remove backslash

File: Library/Homebrew/extend/pathname.rb [hunk 2]
```diff
@@ -371,6 +379,13 @@ def write_exec_script(*targets)
   end
 
   # Writes an exec script that sets environment variables.
+  #
+  # The manner in which the environment variables are set makes them available to
+  # the +target+ when it is executed. However, the environment variables will not
+  # be available for shell parameter expansion within the +target+ string itself.
+  # For example, this method should not be used to +exec ${JAVA_HOME}/bin/java+,
+  # because the +$JAVA_HOME+ from the passed environment will not be available
```
Comment (EricFromCanada) [inline Library/Homebrew/extend/pathname.rb:387]: remove $

File: Library/Homebrew/extend/pathname.rb [hunk 3]
```diff
@@ -354,6 +354,14 @@ def uninstall_info
   end
 
   # Writes an exec script in this folder for each target pathname.
+  #
+  # If the +targets+ require environment variables to be set by the exec script,
+  # then use #write_env_script.
```
Comment (EricFromCanada) [inline Library/Homebrew/extend/pathname.rb:359]: wrap method references in braces to auto-link, e.g. `{#write_env_script}`

**Review by EricFromCanada — CHANGES_REQUESTED** [review]
This project uses Markdown rather than Rdoc, so use backticks to denote fixed-width text.

**Comment by cpurdy** [comment]
Thanks @EricFromCanada! 

I have made the necessary changes according to your instructions.

**Comment by cpurdy** [comment]
Please pay no attention to the man behind the curtain. (The previous commit was a bit smaller than I intended.)

File: Library/Homebrew/extend/pathname.rb [hunk 4]
```diff
@@ -371,6 +379,13 @@ def write_exec_script(*targets)
   end
 
   # Writes an exec script that sets environment variables.
+  #
+  # The manner in which the environment variables are set makes them available to
+  # the `target` when it is executed. However, the environment variables will not
+  # be available for shell parameter expansion within the `target` string itself.
+  # For example, this method should not be used to `exec ${JAVA_HOME}/bin/java`,
+  # because the `JAVA_HOME` from the passed environment will not be available
+  # for shell parameter expansion; use {#write_jar_script} instead.
```
[the remark this answers is not in the document]
Comment (cpurdy) [reply Library/Homebrew/extend/pathname.rb:388]: Let me think about it a bit, and after I digest, I'll make some edits and push back. I'll do some additions to the cookbook separately. (I try to tailor method level comments in a more focused manner, so the approach would be different in a cookbook.)

**Pull request closed without merging**